v1/ast/oracle/oracle.go

```diff
@@ -82,6 +81,31 @@ func (o *Oracle) FindDefinition(q DefinitionQuery) (*DefinitionQueryResult, erro
 		return nil, ErrNoMatchFound
 	}
 
+	// Handle references to rules and imports
+	if result := findRefDefinition(compiler, parsed, stack); result != nil {
+		return result, nil
+	}
+
+	// Handle variable references
+	if result := findVarDefinition(stack); result != nil {
+		return result, nil
+	}
+
+	// Handle some declarations
+	if result := handleSomeDecl(compiler, stack); result != nil {
+		return result, nil
+	}
+
+	// Handle every declarations
+	if result := handleEvery(compiler, stack); result != nil {
+		return result, nil
+	}
```

Comment:
If the number of searches grows here, we might be interested to look into running them at the same time, ordering them such that more complex ones are done last.